apps/ledger-live-desktop/src/renderer/modals/EditTransaction/steps/StepMethod.tsx

```diff
@@ -178,35 +183,36 @@ export class StepMethodFooter extends PureComponent<StepProps> {
                 }),
               );
             } else {
-              updateTransaction(tx =>
+              const mainAccount = getMainAccount(account, parentAccount);
+              updateTransaction((tx: any) =>
                 bridge.updateTransaction(tx, {
                   amount: new BigNumber(0),
                   allowZeroAmount: true,
                   data: undefined,
                   nonce: transactionSequenceNumber,
                   mode: "send",
-                  recipient: account.freshAddress ?? parentAccount.freshAddress,
+                  recipient: mainAccount.freshAddress,
                   // increase gas fees in case of cancel flow as we don't have the fees input screen for cancel flow
                   maxFeePerGas: transactionRaw.maxFeePerGas
                     ? new BigNumber(
                         Math.round(
-                          transactionRaw.maxFeePerGas *
+                          Number(transactionRaw.maxFeePerGas) *
                             (1 + getEnv("EDIT_TX_EIP1559_MAXFEE_GAP_CANCEL_FACTOR")),
```

Comment:
Why not just do

```typescript
transactionRaw.maxFeePerGas.times(....)
```

Reply:
Yup, good catch, also `maxFeePerGas` is supposed to a `uint256`, so can never transform it to a number.

Reply:
fixed